examples/show-errors.rs

```diff
@@ -0,0 +1,68 @@
+use std::borrow::Cow;
+use std::env;
+use std::fs;
+use std::path;
+
+use encoding::all::{ISO_8859_1, UTF_8};
+use encoding::{DecoderTrap, Encoding};
+
+use igc::records::Record;
+
+fn main() {
+    // collect command line arguments
+    let args: Vec<_> = env::args().collect();
+
+    if args.len() < 2 {
+        eprintln!("Usage: show-errors DIR");
+        std::process::exit(1);
+    }
+
+    // iterate over files in the folder
+    for entry in fs::read_dir(&args[1]).unwrap() {
+        let path: path::PathBuf = entry.unwrap().path();
+        if !is_igc_file(&path) {
+            continue;
+        }
+
+        let filename = path.file_name().unwrap().to_str().unwrap();
+
+        // open file in buffered reader
+        let bytes = fs::read(&path).unwrap();
+        let text = match as_text(&bytes) {
```

Comment:
What are the advantages of doing this over using `BufReader::lines`?
Is it that `BufReader` expects utf8 everywhere / doesn't handle different encodings well? (I thought it did, though to be fair have never tried)

Reply:
Yes, exactly. BufReader::lines() reads bytes and internally converts them to Utf8 or returns errors. I'm not aware of any options to make it work with other encodings. Since other encodings are not supported by the stdlib that seems generally unlikely.